ci-operator/step-registry/openstack/provision/sriov-worker/openstack-provision-sriov-worker-commands.sh

```diff
@@ -0,0 +1,144 @@
+            oc get node
+            return 1
+        fi
+    done
+}
+
+# For disconnected or otherwise unreachable environments, we want to
+# have steps use an HTTP(S) proxy to reach the API server. This proxy
+# configuration file should export HTTP_PROXY, HTTPS_PROXY, and NO_PROXY
+# environment variables, as well as their lowercase equivalents (note
+# that libcurl doesn't recognize the uppercase variables).
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+	# shellcheck disable=SC1090
+	source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+if ! openstack network show "${OPENSTACK_SRIOV_NETWORK}" >/dev/null 2>&1; then
+    echo "Network ${OPENSTACK_SRIOV_NETWORK} doesn't exist"
+    exit 1
+fi
+NETWORK_ID=$(openstack network show "${OPENSTACK_SRIOV_NETWORK}" -f value -c id)
+SUBNET_ID=$(openstack network show "${OPENSTACK_SRIOV_NETWORK}" -f json -c subnets | jq '.subnets[0]' | sed 's/"//g')
+
+echo "Scaling down workers to 0"
+WORKER_MACHINESET=$(oc get machineset -n openshift-machine-api | grep worker | awk '{print $1}')
+oc scale --replicas=0 machineset "${WORKER_MACHINESET}" -n openshift-machine-api
+wait_for_no_worker_machines
+
+echo "Downloading current MachineSet for workers"
+oc get machineset -n openshift-machine-api "${WORKER_MACHINESET}" -o json > "${SHARED_DIR}/original-worker-machineset.json"
+cat <<EOF > "${SHARED_DIR}/sriov_patch.json"
+{
+  "spec": {
+    "template": {
+      "spec": {
+        "providerSpec": {
+          "value": {
+            "configDrive": true,
+            "ports": [
+              {
+                "networkID": "${NETWORK_ID}",
+                "nameSuffix": "sriov",
+                "fixedIPs": [
+                  {
+                    "subnetID": "${SUBNET_ID}"
+                  }
+                ],
+                "tags": [
+                  "sriov"
+                ],
+                "vnicType": "direct",
+                "portSecurity": false
+              }
+            ],
+            "trunk": false
+          }
+        }
+      }
+    }
+  }
+}
+EOF
+echo "Merging the original worker MachineSet with the patched configuration for SR-IOV"
+jq -Ss '.[0] * .[1]' "${SHARED_DIR}/original-worker-machineset.json" "${SHARED_DIR}/sriov_patch.json" > "${SHARED_DIR}/sriov-worker-machineset.json"
+python -c 'import sys, yaml, json; yaml.dump(json.load(sys.stdin), sys.stdout, indent=2)' < "${SHARED_DIR}/sriov-worker-machineset.json" > "${SHARED_DIR}/sriov-worker-machineset.yaml"
+
+echo "Apply the new MachineSet for SR-IOV workers"
+oc apply -f "${SHARED_DIR}/sriov-worker-machineset.yaml"
+
+echo "Scaling up worker to 1"
+oc scale --replicas=1 machineset "${WORKER_MACHINESET}" -n openshift-machine-api
+wait_for_worker_machines
+
+echo "Apply SRIOV capable label to the worker node"
+WORKER_NODE=$(oc get node -o custom-columns=NAME:.metadata.name --no-headers -l node-role.kubernetes.io/worker)
+oc label node "${WORKER_NODE}" feature.node.kubernetes.io/network-sriov.capable="true"
+sleep 10
```

Comment:
is this sleep required?

Reply:
yes for now, as we discussed the sleep can help to make sure the label was added to the node. we can improve later.